cmd/grpc/main_integration_test.go

```diff
@@ -0,0 +1,135 @@
+//go:build integration
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"testing"
+	"time"
+
+	"github.com/emmaLP/gs-software-onboarding/internal/caching"
+	"github.com/emmaLP/gs-software-onboarding/internal/config"
+	"github.com/emmaLP/gs-software-onboarding/internal/database"
+	"github.com/emmaLP/gs-software-onboarding/internal/model"
+	commonModel "github.com/emmaLP/gs-software-onboarding/pkg/common/model"
+	pb "github.com/emmaLP/gs-software-onboarding/pkg/grpc/proto"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+type testHandler struct {
+	logger      *zap.Logger
+	config      *model.Configuration
+	dbClient    database.Client
+	cacheClient caching.Client
+}
+
+func TestGrpcServer_ListStories(t *testing.T) {
+	story := commonModel.Item{
+		ID:      1,
+		Dead:    false,
+		Deleted: false,
+		Type:    "story",
+	}
+	story2 := commonModel.Item{
+		ID:      3,
+		Dead:    false,
+		Deleted: false,
+		Type:    "story",
+	}
+	job := commonModel.Item{
+		ID:      2,
+		Dead:    true,
+		Deleted: false,
+		Type:    "job",
+	}
+	tests := map[string]struct {
+		itemsToSave      []*commonModel.Item
+		expectedResponse []*commonModel.Item
+	}{
+		"Successfully list 1 story": {
+			itemsToSave:      []*commonModel.Item{&story, &job},
+			expectedResponse: []*commonModel.Item{&story},
+		},
+		"Successfully list 2 stories": {
+			itemsToSave:      []*commonModel.Item{&story, &job, &story2},
+			expectedResponse: []*commonModel.Item{&story, &story2},
+		},
+	}
+
+	for testName, testConfig := range tests {
+		t.Run(testName, func(t *testing.T) {
+			handler := loadTestHandler(t)
+			ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+			defer cancel()
+			handler.cacheClient.FlushAll(ctx)
+			for _, item := range testConfig.itemsToSave {
+				handler.saveItemToDatabase(t, item)
+			}
+
+			grpcConnection, err := grpc.Dial(handler.address(), grpc.WithInsecure())
+			assert.NoError(t, err)
+			defer grpcConnection.Close()
+
+			client := pb.NewAPIClient(grpcConnection)
+
+			storiesSteam, err := client.ListStories(ctx, &emptypb.Empty{})
+			assert.NoError(t, err)
+			assert.NotNil(t, storiesSteam)
+
+			for _, expectedItem := range testConfig.expectedResponse {
+				out, err := storiesSteam.Recv()
+				if err == nil {
+					actualItem := commonModel.PItemToItem(out)
```

Comment:
Go seems to have a naming convention for converting one model to another where by the shorthand is used. e.g. [strconv.Atoi](https://pkg.go.dev/strconv#Atoi) and [strconv.Itoa](https://pkg.go.dev/strconv#Itoa)

In this case, it could be `commonModel.Ptoi`

Reply:
The `atoi` and `itoa` function are acronyms originally introduced in C language. https://stackoverflow.com/questions/2909768/where-did-the-name-atoi-come-from

Ptoi suggests we are going from P object to an integer